packages/babel-parser/src/parser/statement.ts

```diff
@@ -2638,13 +2656,47 @@ export default abstract class StatementParser extends ExpressionParser {
     }
   }
 
+  parseMaybeImportReflection(node: Undone<N.ImportDeclaration>) {
+    let isImportReflection = false;
+    if (this.isContextual(tt._module)) {
+      const lookahead = this.lookahead();
+      if (tokenIsIdentifier(lookahead.type)) {
+        if (lookahead.type !== tt._from) {
+          // import module x
+          isImportReflection = true;
+        } else {
+          const nextNextTokenFirstChar = this.input.charCodeAt(
+            this.nextTokenStartSince(lookahead.end),
+          );
+          if (nextNextTokenFirstChar === charCodes.lowercaseF) {
+            // import module from from ...
+            isImportReflection = true;
```

Comment:
Can you add a test for `import module from foo`, just to verify that it errors even if `foo` starts with `f`?